plugins/utils/tmux-navigator.nix

```diff
@@ -6,37 +6,173 @@
   ...
 }:
 with lib; let
-  cfg = config.plugins.tmux-navigator;
-in {
-  options.plugins.tmux-navigator = {
-    enable = mkEnableOption "Tmux-Navigator (see https://github.com/christoomey/vim-tmux-navigator for tmux installation instruction)";
+  # TODO: Introduced 2024-03-19, remove on 2024-05-19
+  deprecations = let
+    pluginPath = ["plugins" "tmux-navigator"];
+    option = s: pluginPath ++ [s];
+    setting = s: pluginPath ++ ["settings" s];
+    settingStr = s: concatStringsSep "." (setting s);
+  in [
+    (
+      mkRenamedOptionModule
+      (option "tmuxNavigatorSaveOnSwitch")
+      (setting "save_on_switch")
+    )
+    (
+      mkRemovedOptionModule
+      (option "tmuxNavigatorDisableWhenZoomed")
+      "Use `${settingStr "disable_when_zoomed"}` option."
+    )
+    (
+      mkRemovedOptionModule
+      (option "tmuxNavigatorNoWrap")
+      "Use `${settingStr "no_wrap"}` option."
+    )
+  ];
+in
+  helpers.vim-plugin.mkVimPlugin config {
+    name = "tmux-navigator";
+    originalName = "vim-tmux-navigator";
+    defaultPackage = pkgs.vimPlugins.vim-tmux-navigator;
+    globalPrefix = "tmux_navigator_";
 
-    package = helpers.mkPackageOption "tmux-navigator" pkgs.vimPlugins.tmux-navigator;
+    maintainers = [maintainers.GaetanLepage];
```

Comment:
Wouldn't you want to take ownership of this plugin ?

Reply:
Happy to, it just felt a bit presumptuous. Can't promise active maintainership as time & interest fluctuate

Reply:
Don't worry, this does not have a huge importance.